tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+        feature_name,value that is used to compare against to baseline
+        transposed csv
+        """
+        baseline_ts_json = self.baseline_ts_json
+
+        test_path_created = False
+        if os.path.exists(test_path):
+            test_path_created = True
+        self.assertEqual(test_path_created, True)
+
+        json_data_exists = False
+        if os.path.isfile(baseline_ts_json):
+            json_data_exists = True
+
+        self.assertEqual(json_data_exists, True)
+
+        timeseries_json = None
+        if os.path.isfile(baseline_ts_json):
+            with open(baseline_ts_json, 'r') as f:
+                timeseries_json = json.loads(f.read())
+
+        if python_version == 2:
+            timeseries_str = str(timeseries_json).replace('{u\'results\': ', '').replace('}', '')
+        if python_version == 3:
+            timeseries_str = str(timeseries_json).replace('{\'results\': ', '').replace('}', '')
+
+        full_timeseries = eval(timeseries_str)
+        timeseries = full_timeseries[:60]
+        self.assertEqual(int(timeseries[0][0]), 1369677886)
+        self.assertEqual(len(timeseries), 60)
+
+        for ts, value in timeseries:
+            metric = 'tsfresh_features_test'
+            timestamp = int(ts)
+            value = str(float(value))
+            utc_ts_line = '%s,%s,%s\n' % (metric, str(timestamp), value)
+            with open(tmp_csv, 'a') as fh:
+                fh.write(utc_ts_line)
+
+        if os.path.isfile(tmp_csv):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+
+        df_features = None
+        if os.path.isfile(t_fname_out):
+            return True
+        df = pd.read_csv(tmp_csv, delimiter=',', header=None, names=['metric', 'timestamp', 'value'])
+        df.columns = ['metric', 'timestamp', 'value']
+        df_features = extract_features(df, column_id='metric', column_sort='timestamp', column_kind=None, column_value=None)
+
+        df_created = None
+        df_assert = None
+        # Test the DataFrame
+        try:
+            df_created = str(df_features.head())
+            if df_created:
+                df_assert = True
+                self.assertEqual(df_assert, True)
+        # Catch when df_created is None
+        except AttributeError:
+            self.assertEqual(df_created, True)
+            pass
+        # Catch if not defined
+        except NameError:
+            self.assertEqual(df_created, True)
+            pass
+
+        if os.path.isfile(t_fname_out):
+            return True
+
+        # Write to disk
+        df_features.to_csv(fname_out)
+
+        if os.path.isfile(fname_out):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+
+        if os.path.isfile(t_fname_out):
+            return True
+
+        # Transpose
+        df_t = None
```

Comment:
No but if they are executed in a try/except block and do not exist, then they do not necessarily get declared.  So I try and specifically declared them always so that even if somewhere down the line, something tries them they exist.  I read somewhere that science likes Python as it allows use to “code what we think”.